frontend/src/utils/dataFetching/CRUDposts.js

```diff
@@ -0,0 +1,42 @@
+import { getToken } from "../auth";
+
+// ADD POST
+export const postData = async (url, data) => {
+  const response = await fetch(url, {
+    method: "POST",
+    headers: {
+      Authorization: `Bearer ${getToken()}`,
+      "Content-Type": "application/json"
+    },
+    body: data
+  });
+
+  return response.ok;
+};
+
+// UPDATE POST
+export const updateData = async (url, data) => {
+  const response = await fetch(url, {
+    method: "PATCH",
+    headers: {
+      Authorization: `Bearer ${getToken()}`,
+      "Content-Type": "application/json"
+    },
+    body: data
+  });
+
+  return response.ok;
+};
+
+// DELETE POST
+export const deleteData = async (url, data) => {
+  const response = await fetch(url, {
+    method: "DELETE",
+    headers: {
+      Authorization: `Bearer ${getToken()}`,
+      "Content-Type": "application/json"
+    }
+  });
+
+  return response.ok;
+};
```

Comment:
Why CRUD is in scope of dataFetching - since is not only fetching but also modifying

Reply:
Btw you can have here something like 
```js
export cont async request(action, url, body) => {
  return fetch(url, {
    method: method,
    headers: {
      ....
    }
  })
}

//then

export const deleteData = async (url, data) => {
  const response = await request("DELETE", url, data)
  return response.ok
}

```
and all of other methods